pkg/subctl/cmd/join.go

```diff
@@ -268,16 +270,28 @@ func joinSubmarinerCluster(config *rest.Config, subctlData *datafile.SubctlData)
 		err = checkOverlappingClusterCidr(globalNetworks)
 		status.End(err == nil)
 		exitOnError("Error validating overlapping ClusterCIDRs", err)
-	} else if globalNetworks[clusterID] == nil || globalNetworks[clusterID].GlobalCIDRs == nil || len(globalNetworks[clusterID].GlobalCIDRs) <= 0 {
-		// Globalnet enabled, no globalCidr configured on this cluster
-		globalCIDR, err = globalnet.AllocateGlobalCIDR(globalNetworks, subctlData)
-		status.End(err == nil)
-		status.QueueSuccessMessage(fmt.Sprintf("Allocated GlobalCIDR: %s", globalCIDR))
-		exitOnError("Globalnet failed", err)
+	} else if globalCIDR == "" {
```

Comment:
Rework these checks. Before these checks you should check if user has specified size and cidr both. There are some checks from line 207-213. Can move them to a validateGlobalnetConfiguration() and add these checks. Throw an error that both parameters can't be used together and only one of the two should be used. If using CIDR, it should also validate it is a valid CIDR. Aim is if user supplied invalid input, we should catch it before we do expensive process of discovering networks and doing our checks. Fail fast.

Then check if it is already configured and use existing if present.

Finally handle case where cidr is specified. I think this way it will be easier to read and understand the flow.

Reply:
Added check for invalid CIDR and overlapping CIDR